indexing-service/src/main/java/org/apache/druid/indexing/worker/IntermediaryDataManager.java

```diff
@@ -174,6 +184,19 @@ private void discoverSupervisorTaskPartitions()
           supervisorTaskCheckTimes.computeIfAbsent(
               supervisorTaskId,
               k -> {
+                for (File eachFile : FileUtils.listFiles(supervisorTaskDir, null, true)) {
+                  final String relativeSegmentPath = localtionPath
+                      .relativize(eachFile.toPath().toAbsolutePath())
+                      .toString();
+                  final File reservedFile = location.reserve(
```

Comment:
couldn't understand why we need to do this.

Reply:
Added.